src/openms/source/ANALYSIS/TOPDOWN/FLASHIda.cpp

```diff
@@ -0,0 +1,1127 @@
+        {
+          if (line.hasPrefix("rt"))
+          {
+            continue;
+          }
+          std::stringstream tmp_stream(line);
+          String str;
+          std::vector<String> results;
+          while (getline(tmp_stream, str, '\t'))
+          {
+            results.push_back(str);
+          }
+          mass = atof(results[5].c_str());
+          target_masses_.insert(mass);
+          int nmass = FLASHDeconvAlgorithm::getNominalMass(mass);
+          if (target_nominal_masses_.find(nmass) == target_nominal_masses_.end())
+          {
+            target_nominal_masses_[nmass] = std::vector<double>();
+          }
+          target_nominal_masses_[nmass].push_back(60.0 * atof(results[0].c_str()));
+        }
+        instream.close();
+      }
+    }
+
+    //fd_defaults.setValue("min_mass_count", mass_count_);
+
+    fd_.setParameters(fd_defaults);
+    fd_.calculateAveragine(false);
+    fd_.setTargetMasses(target_masses_, 1);
+
+    std::cout << "QScore threshold: " << qscore_threshold_ << std::endl;
+    std::cout << fd_defaults << std::endl;
+  }
+
+  int FLASHIda::getPeakGroups(const double *mzs,
+                              const double *ints,
+                              const int length,
+                              const double rt,
+                              const int ms_level,
+                              const char *name)
+  {
+
+    auto spec = makeMSSpectrum_(mzs, ints, length, rt, ms_level, name);
+    //deconvoluted_spectrum_ = DeconvolutedSpectrum(spec, 1);
+    if (ms_level == 1)
+    {
+      //current_max_mass_ = max_mass;
+      //currentChargeRange = chargeRange;
+    }
+    else
+    {
+      return 0;
+      //TODO precursor infor here
+    }
+
+    std::vector<DeconvolutedSpectrum> tmp;
+    std::map<int, std::vector<std::vector<double>>> empty;
+    deconvoluted_spectrum_ = fd_.getDeconvolutedSpectrum(spec, tmp, 0, empty);
+
+    // per spec deconvolution
+    FLASHIda::filterPeakGroupsUsingMassExclusion_(ms_level, rt);
+
+    // spec.clear(true);
+
+    return deconvoluted_spectrum_.size();
+  }
+
+  void FLASHIda::filterPeakGroupsUsingMassExclusion_(const int ms_level, const double rt)
+  {
+    std::vector<PeakGroup> filtered_peakgroups;
+
+    //if(next_rt>0 && next_rt > rt){
+    //    deconvoluted_spectrum_.swap(filtered_peakgroups);
+    //    return;
+    //}
+
+    std::sort(deconvoluted_spectrum_.begin(), deconvoluted_spectrum_.end(), QscoreComparator_);
+    int mass_count = mass_count_[ms_level - 1];
+    trigger_charges.clear();
+    trigger_charges.reserve(mass_count);
+
+    filtered_peakgroups.reserve(mass_count_.size());
+    std::set<int> current_selected_mzs; // current selected mzs
+
+    std::unordered_map<int, double> new_mz_rt_map_;
+    std::unordered_map<int, double> new_mass_rt_map_;
+    std::unordered_map<int, double> new_all_mass_rt_map_;
+    std::unordered_map<int, double> new_mass_qscore_map_;
+
+    for (auto &item: mz_rt_map_)
```

Comment:
a bit easier to read with:
for (auto& [m, r] : mz_rt_map)
{
  if (rt - r > rt_window) 
...
}

Reply:
Done